src/pages/experiments/[experimentId]/plots-and-tables/frequency/index.jsx

```diff
@@ -56,45 +57,60 @@ const frequencyPlot = () => {
   }, [experimentId]);
 
   useEffect(() => {
-    if (!config || cellSets.loading) {
+    if (!config || loading) {
       return;
     }
     setPlotSpec(generateSpec(config));
     setPlotData(generateData());
   }, [config, properties]);
 
-  const calculateSum = (chosenClusters) => {
+  const calculateSum = (chosenClusters, metadataIds) => {
     let sum = 0;
-    chosenClusters.forEach((cluster) => {
-      sum += properties[cluster.key].cellIds.size;
+    chosenClusters.map((cellSetCluster) => {
+      const cellSetIds = Array.from(properties[cellSetCluster.key].cellIds);
+      sum += metadataIds.filter((id) => cellSetIds.includes(id)).length;
+      return cellSetIds;
     });
     return sum;
   };
 
+  const getMetadataClusters = (name) => (
+    hierarchy.filter((cluster) => (
+      cluster.key === name))[0].children
+  );
+
   const generateData = () => {
     const data = [];
-    console.log('chosen clusters is ', config.chosenclusters);
-    if (!loading) {
-      const chosenClusters = hierarchy.filter((cluster) => (
-        cluster.key === config.chosenClusters))[0].children;
-      const sum = calculateSum(chosenClusters);
+    const chosenClusters = hierarchy.filter((cluster) => (
+      cluster.key === config.chosenClusters))[0].children;
+
+    const metadataClusters = getMetadataClusters(config.metadata);
+    metadataClusters.map((metadataCluster) => {
+      const metadataIds = Array.from(properties[metadataCluster.key].cellIds);
+      const sum = calculateSum(chosenClusters, metadataIds);
+
       chosenClusters.map((clusterName) => {
         let value;
-        if (config.plotType === 'count') {
-          value = properties[clusterName.key].cellIds.size;
-        } else {
-          value = (properties[clusterName.key].cellIds.size / sum) * 100;
+        const cellSetIds = Array.from(properties[clusterName.key].cellIds);
+        value = metadataIds.filter((id) => cellSetIds.includes(id)).length;
+
+        if (config.plotType === 'proportional') {
```

Comment:
I just noticed that the name `plotType` here is clashing with the `plotType` defined at the top of the class. Rename the `plotType` in `config` to `frequencyType`